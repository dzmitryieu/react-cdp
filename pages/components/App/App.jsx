import React from 'react';
import { hot } from 'react-hot-loader';
import { Switch, Route } from 'react-router-dom';
import Signature from '../Signature';
import MovieSearchPage from '../../MovieSearchPage';
import MovieDetailsPage from '../../MovieDetailsPage';
import PageNotFound from '../../PageNotFound';
import Navigation from '../Navigation';
import MoviesList from '../../MoviesList';
import ErrorBoundary from '../ErrorBoundary';

import './App.scss';

const App = () => (
  <ErrorBoundary>
    <Switch>
      <Route exact path="/" component={MovieSearchPage} />
      <Route exact path="/404" component={PageNotFound} />
      <Route exact path="/film/:id" component={MovieDetailsPage} />
      <Route exact path="/search" component={MovieSearchPage} />
      <Route component={PageNotFound} />
    </Switch>
    <Navigation />
    <MoviesList />
    <Signature />
  </ErrorBoundary>
);

export default hot(module)(App);
