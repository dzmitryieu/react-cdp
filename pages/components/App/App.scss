@import "../../utils/constants.scss";

@font-face {
  font-family: 'Helvetica Neue';
  font-style: normal;
  src: url('../../assets/fonts/HelveticaNeue.woff') format('woff');
}

body {
  margin: 0;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
  font-family: $helvetica;
}

* {
  box-sizing: border-box;
}
