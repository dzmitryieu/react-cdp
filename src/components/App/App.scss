body {
  margin: 0;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

li {
  list-style-type: none;
}

a {
  text-decoration: none;
}
