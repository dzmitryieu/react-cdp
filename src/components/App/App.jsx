import React from 'react';
import { hot } from 'react-hot-loader';
import { Switch, Route } from 'react-router-dom';
import Signature from '../Signature';
import MovieSearchPage from '../../pages/MovieSearchPage';
import MovieDetailsPage from '../../pages/MovieDetailsPage';
import PageNotFound from '../../pages/PageNotFound';
import Navigation from '../Navigation';
import MoviesList from '../../pages/MoviesList';
import ErrorBoundary from '../ErrorBoundary';

import './App.scss';

const App = () => (
  <ErrorBoundary>
    <Switch>
      <Route exact path="/" component={MovieSearchPage} />
      <Route exact path="/404" component={PageNotFound} />
      <Route exact path="/film/:id" component={MovieDetailsPage} />
      <Route exact path="/search" component={MovieSearchPage} />
      <Route component={PageNotFound} />
    </Switch>
    <Navigation />
    <MoviesList />
    <Signature />
  </ErrorBoundary>
);

export default hot(module)(App);
