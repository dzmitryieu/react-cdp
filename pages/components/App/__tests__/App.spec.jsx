import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../index';

const shallow = new ShallowRenderer();

describe('App Snapshot', () => {
  test('App render', () => {
    const component = shallow.render(
      <App />,
    );
    expect(component).toMatchSnapshot();
  });
});
