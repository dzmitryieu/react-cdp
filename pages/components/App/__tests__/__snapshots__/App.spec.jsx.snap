// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot App render 1`] = `
<ErrorBoundary>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/404"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/film/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/search"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Connect(Navigation) />
  <MoviesList />
  <Signature />
</ErrorBoundary>
`;
